src/main/java/edu/harvard/iq/dataverse/engine/command/impl/RegisterOIDCUserCommand.java

```diff
@@ -0,0 +1,128 @@
+package edu.harvard.iq.dataverse.engine.command.impl;
+
+import com.nimbusds.openid.connect.sdk.claims.UserInfo;
+import edu.harvard.iq.dataverse.DvObject;
+import edu.harvard.iq.dataverse.api.dto.UserDTO;
+import edu.harvard.iq.dataverse.authorization.AuthenticatedUserDisplayInfo;
+import edu.harvard.iq.dataverse.authorization.OIDCUserInfo;
+import edu.harvard.iq.dataverse.authorization.UserRecordIdentifier;
+import edu.harvard.iq.dataverse.authorization.exceptions.AuthorizationException;
+import edu.harvard.iq.dataverse.engine.command.*;
+import edu.harvard.iq.dataverse.engine.command.exception.CommandException;
+import edu.harvard.iq.dataverse.engine.command.exception.IllegalCommandException;
+import edu.harvard.iq.dataverse.engine.command.exception.PermissionException;
+import edu.harvard.iq.dataverse.engine.command.exception.InvalidFieldsCommandException;
+import edu.harvard.iq.dataverse.util.BundleUtil;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@RequiredPermissions({})
+public class RegisterOIDCUserCommand extends AbstractVoidCommand {
+
+    private final String bearerToken;
+    private final UserDTO userDTO;
+
+    public RegisterOIDCUserCommand(DataverseRequest aRequest, String bearerToken, UserDTO userDTO) {
+        super(aRequest, (DvObject) null);
+        this.bearerToken = bearerToken;
+        this.userDTO = userDTO;
+    }
+
+    @Override
+    protected void executeImpl(CommandContext ctxt) throws CommandException {
+        try {
+            OIDCUserInfo oidcUserInfo = ctxt.authentication().verifyOIDCBearerTokenAndGetUserIdentifier(bearerToken);
+            UserRecordIdentifier userRecordIdentifier = oidcUserInfo.getUserRecordIdentifier();
+
+            if (ctxt.authentication().lookupUser(userRecordIdentifier) != null) {
+                throw new IllegalCommandException(BundleUtil.getStringFromBundle("registerOidcUserCommand.errors.userAlreadyRegisteredWithToken"), this);
+            }
+
+            UserInfo userClaimsInfo = oidcUserInfo.getUserClaimsInfo();
+
+            // Update the UserDTO object with available OIDC user claims; keep existing values if claims are absent
+            userDTO.setUsername(getValueOrDefault(userClaimsInfo.getPreferredUsername(), userDTO.getUsername()));
+            userDTO.setFirstName(getValueOrDefault(userClaimsInfo.getGivenName(), userDTO.getFirstName()));
+            userDTO.setLastName(getValueOrDefault(userClaimsInfo.getFamilyName(), userDTO.getLastName()));
+            userDTO.setEmailAddress(getValueOrDefault(userClaimsInfo.getEmailAddress(), userDTO.getEmailAddress()));
```

Comment:
Could we also try to use the[ ` email_verified`]( https://openid.net/specs/openid-connect-core-1_0.html#AddressClaim)?

Reply:
This is a feature that is currently only available for Shibboleth and not that visable. 

https://github.com/IQSS/dataverse/blob/f838e76baa055243a80331be57543060ae25e6a9/src/main/java/edu/harvard/iq/dataverse/authorization/AuthenticationServiceBean.java#L635